File: program/find-factorial-of-a-number-by-recursion/FindFactorialOfANumberByRecursion.ts
function factorialize(num: number): number {
  if (num < 0) {return -1}
    
  // If the number is 0, its factorial is 1.
  else if (num == 0) {return 1}
    
  // Otherwise, call the recursive procedure again
    else {
        return (num * factorialize(num - 1));
    }
}

factorialize(5);

console.log("Factorial of 5 is", factorialize(5))

File: program/find-factorial-of-a-number-by-recursion/find_factorial_of_a_number_by_recursion.go
package main

import "fmt"

func factorialize(num int) int {
	if (num < 0) {
		return -1;
	} else if (num == 0) {
		return 1
	} else {
		return num * factorialize(num - 1)
	}
}

func main() {
	var i int
	fmt.Print("Enter Number: ")
	fmt.Scan(&i)
	fmt.Println(factorialize(i))
}
